src/bluefinClient.ts

```diff
@@ -2008,4 +2019,60 @@ export class BluefinClient {
       throw new Error(error);
     }
   };
+
+  /**
+   * @param to recipient wallet address
+   * @param balance SUI amount to transfer
+   * @returns Response Schema
+   * @description
+   * transfer sui tokens
+   * */
+  transferSuiBalance = async (
+    to: string,
+    balance: number
+  ): Promise<ResponseSchema> => {
+    return this.contractCalls.transferSuiBalance(to, balance);
+  };
+
+  /**
+   * @param to recipient wallet address
+   * @param balance USDC amount to transfer
+   * @returns Response Schema
+   * @description
+   * transfer USDC tokens
+   * */
+  transferUSDC = async (
+    to: string,
+    balance: number
+  ): Promise<ResponseSchema> => {
+    return this.contractCalls.transferUSDC(to, balance);
+  };
+
+  /**
+   * @param to recipient wallet address
+   * @param balance amount to transfer
+   * @returns BigInt
+   * @description
+   * estimate gas for sui token transfer
+   * */
+  estimateGasForSuiTransfer = async (
```

Comment:
please make a single method for estimation of gas that should take a transaction block as input, dry runs it and returns the estimated gas fee amount.

Reply:
I already declared a utility estimateGas in library sui that takes tx as an argument and returns the estimated gas. But I exposed two separate utilities for estimation of gas for sui and usdc transfer.

If I need to expose estimate gas utility then I also need to expose prepareTransactionForUSDCTransfer & prepareTransactionForSUITransfer and call them on FE when input change and pass it to estimateGas utility. I that case we have to expose three utilities

- estimateGas
- prepareTransactionForUSDCTransfer
- prepareTransactionForSUITransfer

That's why decided to expose estimateGasForUSDCTransfer & estimateGasForSUITransfer which is calling estimateGas and prepareTrasansaction internally.